guidelines/techpreview.md

```diff
@@ -0,0 +1,97 @@
+# What does it mean to be Tech Preview
+
+* You can change APIs that are clearly indicated as tech preview, without following a deprecating or backwards compatibility process.
```

Comment:
This is dev-oriented phrasing.  But users can read this too, and will also care about what  Tech Preview means.  Can we land this in openshift-docs instead?  Hunting earlier, I was unable to find a definition there (#114).

Reply:
a lot of this is not user facing.  so yeah, we can certainly put some of this in the docs once it's agreed upon, but i think we'll need something slightly different for docs vs what we want here for engineers to understand what they need to do.